galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java

```diff
@@ -14,16 +42,241 @@
 "osgi.http.whiteboard.servlet.pattern=/ras/run" }, name = "Galasa Runs microservice")
 public class RunQuery extends HttpServlet {
 	
+	@Reference
+	IFramework framework;
 	
 	private static final long serialVersionUID = 1L;
 	
 	@Override
-	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException {
+	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+		
+		Gson gson = new Gson();
+		int pageNum = -1;
+		
+		Map<String, String> paramMap = getParameterMap(req);
+		
+		List<IRasSearchCriteria> critList = new ArrayList<>();
+		
+		if(!paramMap.isEmpty()) {
+		
+			String requestor = paramMap.get("requestor");
+			String to = paramMap.get("to");
+			String from = paramMap.get("from");
+			String testName = paramMap.get("testName");
+			
+			Instant toCrit = null;
+			Instant fromCrit = null;
+		try {
+			if(to != null) {
+				toCrit = Instant.parse(to);
+				RasSearchCriteriaQueuedTo toCriteria = new RasSearchCriteriaQueuedTo(toCrit);
+				critList.add(toCriteria);
+			}
+			if(from != null) {
+				fromCrit = Instant.parse(from);
+				RasSearchCriteriaQueuedFrom fromCriteria = new RasSearchCriteriaQueuedFrom(fromCrit);
+				critList.add(fromCriteria);
+			}
+		}catch(Exception e) {
+			
+			resp.setStatus(400);
+			PrintWriter out = resp.getWriter();
+			resp.setContentType( "Application/json");
+			resp.setHeader("Access-Control-Allow-Origin", "*");
+			
+			JsonError error = new JsonError("Error parsing Instant");
+			
+			String jsonError = gson.toJson(error);
+			
+			out.print(jsonError);
+			out.flush();
+			
+			throw new ServletException("Error parsing Instant, ", e);
+		}
+			if(requestor != null) {
+				RasSearchCriteriaRequestor requestorCriteria = new RasSearchCriteriaRequestor(requestor);
+				critList.add(requestorCriteria);
+			}
+			if(testName != null) {
+				RasSearchCriteriaTestName testNameCriteria = new RasSearchCriteriaTestName(testName);
+				critList.add(testNameCriteria);
+			}
+
+			
+		}
+		
+		List<RunResult> runs = new ArrayList<>();
+		
+		try {
+			runs = getRuns(critList);
+		} catch (ResultArchiveStoreException e) {
+			e.printStackTrace();
+		}
+		
+		
+		runs.sort(Comparator.nullsLast(Comparator.comparing(RunResult::getEnd, Comparator.nullsLast(Comparator.naturalOrder()))));
+		
+		Map<String, String[]> query = req.getParameterMap();
+		
+		if(!query.isEmpty()){
+			if(!ExtractQuerySort.isAscending(query, "to")) {
+				runs.sort(Comparator.nullsLast(Comparator.comparing(RunResult::getEnd, Comparator.nullsLast(Comparator.naturalOrder()))).reversed());
+			}
+		}
+		
+		int pageSize = 100;
+		
+		if(paramMap.get("pageSize") != null && !paramMap.get("pageSize").equals("")) {
```

Comment:
Get all the parameters at the top of the method, so it is clear where it is coming from,  and what is being defaulted